sharding-core/sharding-core-merge/src/main/java/org/apache/shardingsphere/core/merge/dal/show/ShowDatabasesMergedResult.java

```diff
@@ -30,22 +31,42 @@
 import java.sql.SQLXML;
 import java.util.Collections;
 import java.util.List;
+import org.apache.shardingsphere.core.rule.ShardingRule;
 
 /**
  * Merged result for show databases.
  *
  * @author chenqingyang
  */
-@RequiredArgsConstructor
 public final class ShowDatabasesMergedResult extends LocalMergedResultAdapter implements MergedResult {
     
     private final List<String> schemas;
     
     private int currentIndex;
-    
+
     public ShowDatabasesMergedResult() {
         this(Collections.singletonList(ShardingConstant.LOGIC_SCHEMA_NAME));
     }
+
+    //for proxy backend, just return logic database
+    public ShowDatabasesMergedResult(final List<String> schemas) {
+        this.schemas = schemas;
+        this.currentIndex = 0;
+    }
+
+    public ShowDatabasesMergedResult(final ShardingRule shardingRule, final List<QueryResult> queryResults)throws SQLException {
+        this.schemas = new ArrayList<>();
+        this.currentIndex = 0;
+        init(queryResults);
+    }
+
+    private void init(final List<QueryResult> queryResults) throws SQLException {
+        for (QueryResult queryResult : queryResults) {
+            while (queryResult.next()) {
+                this.schemas.add((String) queryResult.getValue(1, String.class));
```

Comment:
Maybe we can remove unnecessary `this`, IDEA can help us to recognize global variable and local variable

Reply:
refactor it... done